core/src/main/java/io/jenkins/pluginhealth/scoring/probes/ProbeContext.java

```diff
@@ -88,6 +89,14 @@ public Optional<String> getRepositoryName(String scm) {
         return match.find() ? Optional.of(match.group("repo")) : Optional.empty();
     }
 
+    public Optional<Path> getScmFolderPath() {
```

Comment:
I changed the return type to `Optional<Path>` I hope it is okay. Adrien said in the meeting he wants it to be `String`. Would this make a big difference? 

Also, it aligns with the method name like `getScmFolderPath` returns a `Path` which is one of the things you always tell me to focus on: naming methods and conventions.

Reply:
Yeah at first (without playing with it a bit to check edge cases), I would be tempted to say it's better to keep the object as a `String`. 

We really do just want to get the folders that are there, and a `Path` object can contain more.

The fact that the name of the field is `ScmFolderPath` is because we indeed return the folder path, it should not have something to do with the Java type `Path`. Introducing the object type in the variable name is not that great of a practice, even though it was done for years.